elyra/pipeline/component.py

```diff
@@ -418,49 +416,80 @@ def _format_description(self, description: str, data_type: str) -> str:
             return f"{description} (type: {data_type})"
         return f"(type: {data_type})"
 
-    def determine_type_information(self, parsed_type: str) -> Tuple[str, str, Any]:
+    def determine_type_information(self, parsed_type: str) -> SimpleNamespace:
         """
         Takes the type information of a component parameter as parsed from the component
         specification and returns a new type that is one of several standard options.
 
         """
-        type_lowered = parsed_type.lower()
-        type_options = ['dictionary', 'dict', 'set', 'list', 'array', 'arr']
+        parsed_type_lowered = parsed_type.lower()
 
-        # Prefer types that occur in a clause of the form "[type] of ..."
+        data_type_info: SimpleNamespace
+
+        # Determine if this is a "container type"
+        # Prefer types that occur in a clause of the form "[type] of ..." (i.e., "container" types)
         # E.g. "a dictionary of key/value pairs" will produce the type "dictionary"
-        for option in type_options:
-            if any(word + " of " in type_lowered for word in type_options):
-                reg = re.compile(f"({option}) of ")
-                match = reg.search(type_lowered)
-                if match:
-                    type_lowered = option
-                    break
-            elif option in type_lowered:
-                type_lowered = option
+        container_types = ['dictionary', 'dict', 'set', 'list', 'array', 'arr']
+        for option in container_types:
+            if option in parsed_type_lowered:
```

Comment:
Love this idea! Looks and feels so much cleaner. 

I really only have one thought -- I don't know how often we might come across either `list of dicts` or `set of lists`, e.g., in a `parsed_type_lowered`, but I think those cases were the basis for the original implementation of the `determine_type_information` function

Reply:
Good point. 
> basis for the original implementation of the determine_type_information function

By this, are you referring to the nesting of the for loops where the option in "container_types" was the outer loop then the types were checked again?  Or is this in reference to the "if 'of' check, then the plain elif" blocks?

Could/should we move the "dictionary" entries to the end of the list to handle this, since all other entries could contain dictionaries?  Or I suppose we could run across a description like "dictionary of lists", but the key/value aspect usually needs to be addressed - so perhaps that is less of an issue.

If we did encounter "list of dictionaries" or "array of dictionaries", which data type should be determined?  (First, that is really complicated so I would hope that doesn't occur in nature, but I still think you'd need to determine 'list' here.)

Are all "subtypes" (i.e., what the 'of' refers to in the expression) assumed to be strings?  Since we don't really have a way to determine "list of integers", I'm assuming, in that case, we still want a "list" to be the result and not an "integer" - correct?  And do we have the ability to express "of-types" to the front-end?